orangetreedinner.rb

```diff
@@ -0,0 +1,146 @@
+# Orange Tree
+# group: Alice, Elia, Jeri
+
+class OrangeTree
+	attr_reader :age, :alive, :grove, :height, :orange_count
+
+	def initialize
+		@age = 0
+		@alive = true
+		@height = 0
+		@orange_count = 0
+	end
+
+	# one year passes on one tree- main method call- calls private methods
+	def one_year_passes
+		grow_older
+		if @alive == true
+			grow_taller
+			grow_oranges
+		end
+
+		if @grove
+			@grove.soil_quality
+		end
+	end
+
+	def pick_orange
+		if @orange_count != 0
+			@orange_count -= 1
+			puts "That was a very tasty orange."
+
+		else
+			puts "There are no more oranges to pick this year."
+		end
+	end
+
+	def plant_on(grove_name)
+		grove_name.trees.push(self)
+		@grove = grove_name
+	end
```

Comment:
Nice, I like that you create a relationship between the tree and the grove on this side, too.

I think the instructions want you to make this a class method that creates a new instance of `OrangeTree` rather than associating an existing tree instance with a grove. You might also want to give `initialize` the responsibility of keeping track of the grove the tree belongs to. If you don't set it when making a new tree, `@grove` will be `nil` and your logic in your other methods should still work (`nil` evaluates to falsey in Ruby).